core/src/verified_vote_packets.rs

```diff
@@ -229,22 +238,32 @@ impl VerifiedVotePackets {
                     }
                     let slot = vote.last_voted_slot().unwrap();
                     let hash = vote.hash();
+                    let timestamp = vote.timestamp();
 
                     match (vote, is_full_tower_vote_enabled) {
                         (VoteStateUpdate(_), true) => {
-                            let latest_gossip_slot = match self.0.get(&vote_account_key) {
-                                Some(vote) => vote.get_latest_gossip_slot(),
-                                _ => 0,
-                            };
+                            let (latest_gossip_slot, latest_timestamp) =
+                                match self.0.get(&vote_account_key) {
+                                    Some(vote) => (
+                                        vote.get_latest_gossip_slot(),
+                                        vote.get_latest_gossip_timestamp(),
+                                    ),
+                                    _ => (0, None),
+                                };
                             // Since votes are not incremental, we keep only the latest vote
-                            if slot > latest_gossip_slot {
+                            // If the vote is for the same slot we will only allow it if
+                            // it has a later timestamp (refreshed vote)
+                            if slot > latest_gossip_slot
+                                || ((slot == latest_gossip_slot) && (timestamp > latest_timestamp))
```

Comment:
any concerns with these timestamps being wrapped in `Option`?  a `Some` is always greater than a `None`

Reply:
Was intentional, votes are sent without timestamps if something is wrong (`maybe_timestamp` in `consensus.rs`). Figured they should be the lowest priority when breaking ties. 
Could also do an `unwrap_or_default` before comparing if it's more readable.

Reply:
cool, thanks for the details!  i think a comment would suffice. `.unwrap_or_default()` is the devil

Reply:
Suppose we have two forks going, `F1` and `F2`. If we were voting on `F2` and then decided to switch to `F1`, we would NOT ever try to vote on a slot in `F1` that was less than a slot we had voted on in `F2` right? I don't think it would be, but just double checking while we all already have eyes on this bit of logic.

Reply:
yeah in general we can't vote on a slot less than the previous vote because it would violate the lockouts in tower. `F2` being in the tower means we can only vote on slots greater than `F2` / greater than the largest lockout if we are switching.

Reply:
Got it, didn't think so but figured I'd double check with an expert real quick 😄